homeassistant/config.py

```diff
@@ -736,6 +736,11 @@ def async_process_component_config(
                 async_log_exception(ex, domain, p_config, hass)
                 continue
 
+            # Allow PLATFORM_SCHEMA_BASE modify or inject platform name
+            # To avoid breaking change when we have to change platform name
+            # For example google/tts => google_translate/tts
+            p_name = p_validated.get(CONF_PLATFORM)
```

Comment:
My main concern here is that now the TTS integration needs to know about renames of other integrations, that seems weird.